lib/streamlit/elements/vega_charts.py

```diff
@@ -953,20 +1033,50 @@ def scatter_chart(
             width=width,
             height=height,
         )
-        return self._altair_chart(
-            chart,
-            use_container_width=use_container_width,
-            theme="streamlit",
-            add_rows_metadata=add_rows_metadata,
+        return cast(
+            "DeltaGenerator",
+            self._altair_chart(
+                chart,
+                use_container_width=use_container_width,
+                theme="streamlit",
+                add_rows_metadata=add_rows_metadata,
+            ),
         )
 
-    @gather_metrics("altair_chart")
+    @overload
     def altair_chart(
         self,
         altair_chart: alt.Chart,
+        *,
```

Comment:
will this break existing apps that use `altair_chart(chart, False)`? in this case, the `use_container_width` would need to be passed by keyword argument whereas it would not need to be passed like that before, correct?

Reply:
Yep, that's intentional. There are only 4 tracked apps that use it positional so we are good to do this since we anyways want to have almost all parameters keyword only.

<img width="434" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/9ec9cedb-06a4-4b72-b07d-74104b8a26fd">